x/auth/src/ante/ante.rs

```diff
@@ -275,21 +284,60 @@ impl<BK: BankKeeper<SK>, AK: AuthKeeper<SK>, SK: StoreKey> BaseAnteHandler<BK, A
                 )));
             }
 
-            // check signature
-            let sign_bytes = SignDoc {
-                body_bytes: tx.raw.body_bytes.clone(),
-                auth_info_bytes: tx.raw.auth_info_bytes.clone(),
-                chain_id: ctx.get_chain_id().to_owned(),
-                account_number: acct.get_account_number(),
-            }
-            .encode_to_vec();
-            let message = secp256k1::Message::from_hashed_data::<sha256::Hash>(&sign_bytes);
-
             let public_key = acct
                 .get_public_key()
                 .as_ref()
                 .expect("account pub keys are set in set_pub_key_ante_handler"); //TODO: but can't they be set to None?
 
+            let sign_bytes = match &signature_data.mode_info {
+                ModeInfo::Single(mode) => match mode {
+                    SignMode::Unspecified => {
+                        return Err(AppError::TxValidation(
+                            "unspecified sign mode not supported".to_string(),
+                        ));
+                    }
+                    SignMode::Direct => SignDoc {
+                        body_bytes: tx.raw.body_bytes.clone(),
+                        auth_info_bytes: tx.raw.auth_info_bytes.clone(),
+                        chain_id: ctx.get_chain_id().to_owned(),
+                        account_number: acct.get_account_number(),
+                    }
+                    .encode_to_vec(),
+                    SignMode::Textual => {
+                        let _handler = SignModeHandler;
+
+                        let _signer_data = SignerData {
+                            address: signer.to_owned(),
+                            chain_id: ChainId::new(ctx.get_chain_id().to_owned()).unwrap(), //TODO: remove unwrap
+                            account_number: acct.get_account_number(),
+                            sequence: account_seq,
+                            pub_key: public_key.to_owned(),
+                        };
+
+                        let _tx_data = TxData {
+                            body: tx.tx.body.clone(),
+                            auth_info: tx.tx.auth_info.clone(),
+                        };
+
+                        //handler.sign_bytes_get(ctx, signer_data, tx_data).unwrap()
+                        //TODO: remove unwrap
+                        todo!()
```

Comment:
Is implementing this code should be part of this PR?

Reply:
This PR is getting a bit big. So I'll do that as a separate PR.